zcash_client_sqlite/src/wallet.rs

```diff
@@ -2115,10 +2123,13 @@ pub(crate) fn get_unspent_transparent_output(
     outpoint: &OutPoint,
 ) -> Result<Option<WalletTransparentOutput>, SqliteClientError> {
     let mut stmt_select_utxo = conn.prepare_cached(
-        "SELECT u.prevout_txid, u.prevout_idx, u.script, u.value_zat, u.height
-         FROM utxos u
-         WHERE u.prevout_txid = :txid
-         AND u.prevout_idx = :output_index
+        "SELECT t.txid, u.output_index, u.script,
+                u.value_zat, t.mined_height AS received_height
+         FROM transparent_received_outputs u
+         JOIN transactions t ON t.id_tx = u.transaction_id
+         WHERE t.txid = :txid
+         AND t.mined_height IS NOT NULL
+         AND u.output_index = :output_index
          AND u.id NOT IN (
             SELECT txo_spends.transparent_received_output_id
             FROM transparent_received_output_spends txo_spends
```

Comment:
```
OR tx.expiry_height IS NULL -- the spending tx will not expire
```
It could be that the spending tx is unmined and will not expire, but is dependent on another transaction that is unmined and *will* expire.